composer/chain-state.ipynb

```diff
@@ -0,0 +1,174 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Where do you edit this source? I imagine for now it happens in Jupyter and we somehow export it? Asking, because I see this is Python that is somehow JSON marshalled.

Reply:
I wrote this in the Jupyter UI, which isn't the nicest editor but is good for exploring. I think I'll pull it into a separate `.py` file and import it into the notebook soon though, since it's hard to read diffs of python in JSON.